superduperdb/db/base/cdc.py

```diff
@@ -45,39 +327,113 @@ class DatabaseListenerFactory(t.Generic[DBListenerType]):
     def __init__(self, db_type: str = 'mongodb'):
         if db_type not in self.SUPPORTED_LISTENERS:
             raise ValueError(f'{db_type} is not supported yet for CDC.')
-        self.listener = db_type
+        self.db_type = db_type
 
     def create(self, *args, **kwargs) -> DBListenerType:
         stop_event = Event()
         kwargs['stop_event'] = stop_event
-        listener = MongoDatabaseListener(*args, **kwargs)
-        return t.cast(DBListenerType, listener)
+        if self.db_type == 'mongodb':
+            from superduperdb.db.mongodb.cdc.listener import MongoDatabaseListener
+
+            listener = MongoDatabaseListener(*args, **kwargs)
+            return t.cast(DBListenerType, listener)
+        else:
+            raise NotImplementedError
 
 
-def DatabaseListener(
-    db: DB,
-    on: Collection,
-    identifier: str = '',
-    *args,
-    **kwargs,
-) -> BaseDatabaseListener:
+class DatabaseChangeDataCapture:
     """
-    Create an instance of ``BaseDatabaseListener``.
-    Not to be confused with ``superduperdb.container.listener.Listener``.
+    DatabaseChangeDataCapture is a Python class that provides a flexible and
+    extensible framework for capturing and managing data changes
+    in a relational database.
 
-    :param db: A superduperdb instance.
-    :param on: Which collection/table listener service this be invoked on?
-    :param identifier: A identity given to the listener service.
+    This class is repsonsible for cdc service on the provided `db` instance
+    This class is designed to simplify the process of tracking changes
+    to database records,allowing you to monitor and respond to
+    data modifications efficiently.
     """
-    it = backends.data_backends.items()
-    if types := [k for k, v in it if isinstance(db.databackend, v)]:
-        db_type = types[0]
-    else:
-        raise ValueError('No backends found')
 
-    if db_type != 'mongodb':
-        raise NotImplementedError(f'Database {db_type} not supported yet!')
+    def __init__(self, db: 'DB'):
+        self.db = db
+        self._cdc_stop_event = Event()
+        self.CDC_QUEUE: queue.Queue = queue.Queue()
+        self.cdc_change_handler: t.Optional[CDCHandler] = None
+        self._CDC_LISTENERS: t.Dict[str, BaseDatabaseListener] = {}
+        self._running: bool = False
+        self._cdc_existing_collections: t.MutableSequence['TableOrCollection'] = []
```

Comment:
How is this connected with the developer's or user's configurations?

Reply:
I think we should be appending to this every time a `container.listener.Listener` is created ***if***
`s.CFC.cdc == True`.

Reply:
WDYT?